h2/src/main/org/h2/fulltext/FullTextSettings.java

```diff
@@ -10,20 +10,22 @@
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
-import java.util.HashMap;
-import java.util.HashSet;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
 import org.h2.util.New;
 import org.h2.util.SoftHashMap;
 
 /**
  * The global settings of a full text search.
  */
-class FullTextSettings {
+final class FullTextSettings {
 
     /**
      * The settings of open indexes.
      */
-    private static final HashMap<String, FullTextSettings> SETTINGS = New.hashMap();
+    private static final Map<String, FullTextSettings> SETTINGS = Collections.synchronizedMap(New.<String, FullTextSettings>hashMap());
 
```

Comment:
That was the first time I've used Collections.synchronized..() myself, so I am with you on that one. On the other hand I'd never come up with API like this - getter returning internal modifiable collection.
In that particular case there are no iterations, but I did miss a couple of "put-If-Absent" 8-(